### DISCUSSION TITLE
投稿の表示をマークダウンにしませんか？

### DISCUSSION THREAD

**Discussion by Hasuro-R** [Futurealize-with-Tech/fwt#44] [Ideas]
## 投稿をマークダウン対応にしたい！
よく考えてみると、改行などを繰り返された時にはどうしてもカードから見切れてしまう問題は発生してしまう。

だったら、初めからそれを対策し、その上でマークダウン対応にするのは面白いんじゃないか？と考えた次第です。

## 理由
- ただの文字列よりも読みやすい (自分でタイトルもつけられる)
- これをきっかけに、マークダウンについて知ってほしい (エンジニアには割と必須知識)

## 実装するなら
- もしやるなら、マークダウンについて知れるために、ちょっとした**チュートリアル**はつける。
- エディタはまさにこの`GitHub`のエディタのようなイメージ

WYSIWYGエディタではなく、自分でマークダウンを書く方式にしたいのは、これきっかけでマークダウンをかけるメンバーが増えてほしいから。

## 前提条件
勿論、エディタやチュートリアルなど少し実装量は増えてしまうので、**余裕があれば**のお話。

今決めなくていいのですが、忘れないうちに書き記しておきました。

**Comment by Hasuro-R** [comment]
文字が長すぎる時に画像化する解決案

以下のような流れで解決できるかも？

1. 文字の長さによってフォントサイズを変更
2. その本文を画像に変換
3. カードの本文のスペースに、文字画像を`height: 100%`で表示

こうすれば、画像からはみ出ることなく表示できるはず（普通にみる時はスクロールできるようにすれば良い）

**Reply by Hasuro-R** [reply]
ただ、これやとめっちゃ改行したり文字数多い時は、文字めっちゃ小さくなるからどうしようって感じではある

**Reply by xx-n59** [reply]
マークダウン→次繋ぐ時までに考えておきます

画像化って表示する時に既に文字部分だけ画像化されてるってこと？それともダウンロードとか実装する時にってこと？

表示する時はスクロールの方が良いと思う、ダウンロードできるようにするのなら画像化して貼り付けるしかないと思う

**Reply by Hasuro-R** [reply]
本文を画像化して表示するのは、画像として保存する時の解決案やで
普通に表示するときにいちいち画像化するのは処理重いし見にくいから

画像としてダウンロードする時はスクロールやと見切れちゃうから、その対策としてって感じ
（ただ、画像として保存するときは、勿論文字が小さくなって見えにくくなる）

**Reply by xx-n59** [reply]
理解！画像として保存するときはその方がいいと思う
そこの処理はこの前やったので書きます

文字数制限一応加えておいた方がいいかなと思うんだけどどう思う？

**Reply by Hasuro-R** [reply]
ちょうど今下にコメントした💦

**Comment by Hasuro-R** [comment]
> 文字が長すぎる時に画像化する解決案
> 
> 以下のような流れで解決できるかも？
> 
> 1. 文字の長さによってフォントサイズを変更
> 2. その本文を画像に変換
> 3. カードの本文のスペースに、文字画像を`height: 100%`で表示
> 
> こうすれば、画像からはみ出ることなく表示できるはず（普通にみる時はスクロールできるようにすれば良い）

ただ、これの問題点としては、文字数を制限する必要がある（じゃないと文字が小さくなりすぎる）

メッセージカードにかけるメッセージを制限するのはちょっと寂しい気もするし（勿論無限に投稿できないように制限は必須やけど）、Twitterみたいにスレッドを立てれるわけじゃないから本当に制限ないでしか書けない

保存する時に分割するってのもありやけど、そうなると今度は認定証っていうコンセプトから若干変わってしまう


まぁ、文字がアホみたいに小さくても拡大すれば一応読めることは読めるけど、どうしようかって感じ

**Reply by xx-n59** [reply]
分割は絶対やだ

今スマホの方で認定証のデザインを表示して文字入力してみたんだけど意外と文字数は稼げると思う(200字くらいなら余裕で見える)

それか、文字数によってデザインかえるとかもありだと思う
壁紙サイズくらいの認定証ならさすがに入るだろうし

**Reply by Hasuro-R** [reply]
分割は処理むずそうやし、あんまり採用するメリットないと思う

ただ200やと多分少ないし、改行めっちゃしてたら文字数多くなくても小さくなるからデザイン分けても小さくなる可能性はある

けど、デザインというか、ダウンロード形式を２つ用意すれば良いと思った！

- カードのデザインで保存（デザイン重視）
- 本文のみを保存（文字重視）

カードデザインで十分ならそれで良いし、読みにくいなと思った人は、本文とメンバー名だけ書かれた背景に制限がないものをダウンロードすればいいと思う！（勿論、本文のみの方にも背景色薄くつけたりレイアウト整えて、ただの文字っぽくないようにはするつもり）

これならコンセプトも守れて、文字数も制限せず、かつ後から読みやすいからUXも悪くならない気がする

**Reply by xx-n59** [reply]
確かに、一旦それで行こう
